scvi/model/_utils.py

```diff
@@ -10,10 +11,18 @@
 from scvi import REGISTRY_KEYS
 from scvi._types import Number
 from scvi.data import AnnDataManager
+from scvi.data._built_in_data._download import _download
 
 logger = logging.getLogger(__name__)
 
 
+def _download_if_missing(filepath: str, backup_url: Optional[str]):
```

Comment:
this function might not work in some cases where you can't simply add the filename to the url. I think we should remove backup url from the new legacy function and in the new load function, backup url should download the file with the user provided url and put the file inside the load path specified by the user (just like how backup url in scanpy read works)

Reply:
I think this situation arises in zenodo for example